frontend/src/components/bs-space/bs-space.tsx

```diff
@@ -0,0 +1,108 @@
+import { IconName } from 'common/enums/enums';
+import { IconButton } from 'components/common/common';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+  useParams,
+} from 'hooks/hooks';
+import { ObjectsTable } from './components/components';
+import React, { FC } from 'react';
+import styles from './styles.module.scss';
+import { BSSpace as BSSpaceActions } from 'store/actions';
+
+const BSSpace: FC = () => {
+  const objects = useAppSelector((state) => state.BSSpace.objects);
+  const dispatch = useAppDispatch();
+
+  const { id } = useParams();
+
+  useEffect(() => {
+    dispatch(
+      BSSpaceActions.loadObjects({
+        filter: {
+          from: 0,
+          count: 5,
+        },
+        id: id as string,
+      }),
+    );
+  }, [dispatch]);
+
+  const handleObjectsReload = (): void => {
+    dispatch(
+      BSSpaceActions.loadObjects({
+        filter: {
+          from: 0,
+          count: 5,
+        },
+        id: id as string,
+      }),
+    );
+  };
+
+  const handleObjectDownload = (objectId: string): void => {
+    const filename = getFilenameById(objectId);
+
+    if (!filename) {
+      return;
+    }
+
+    dispatch(
+      BSSpaceActions.downloadObject({
+        filename,
+        spaceId: id as string,
+        objectId,
+      }),
+    );
+  };
+
+  const handleObjectUpload = (e: React.FormEvent<HTMLInputElement>): void => {
+    const files = e.currentTarget.files;
```

Comment:
isn't fixed

Reply:
ok, I will add hasFiles because you've asked me about it for the second time, but why do we need redundant variable?🤔